Implement "check-latest" flag to check if pre-cached version is latest one (#186)
actions/setup-go: README.md

The V2 offers:
- Adds GOBIN to the PATH
- Proxy Support
- `stable` input 
- Check latest version
- Bug Fixes (including issues around version matching and semver)

> It will first check the local cache for a version match. If version is not found locally, It will pull it from `main` branch of [go-versions](https://github.com/actions/go-versions/blob/main/versions-manifest.json) repository and on miss or failure, it will fall back to the previous behavior of download directly from [go dist](https://storage.googleapis.com/golang). To change the default behaviour please use [check-latest input](#check-latest-version).

Reviewer: ```suggestion
The action will first check the local cache for a version match. If a version is not found locally, it will pull it from the `main` branch of the [go-versions](https://github.com/actions/go-versions/blob/main/versions-manifest.json) repository and on miss or failure, will fall back to the previous behavior of downloading directly from [go dist](https://storage.googleapis.com/golang). To change the default behaviour please use the [check-latest input](#check-latest-version).
```

Author: Done. Thank you.

Author: Done. Thank you.

> The action will first check the local cache for a version match. If a version is not found locally, it will pull it from the `main` branch of the [go-versions](https://github.com/actions/go-versions/blob/main/versions-manifest.json) repository and on miss or failure, will fall back to the previous behavior of downloading directly from [go dist](https://storage.googleapis.com/golang). To change the default behaviour please use the [check-latest input](#check-latest-version).

Reviewer 2: ```suggestion
The action will first check the local cache for a version match. If a version is not found locally, it will pull it from the `main` branch of the [go-versions](https://github.com/actions/go-versions/blob/main/versions-manifest.json) repository. On miss or failure, it will fall back to downloading directly from [go dist](https://storage.googleapis.com/golang). To change the default behavior, please use the [check-latest input](#check-latest-version).
```

Thanks for updating the README!  A small edit: I don't think it makes sense to refer to "the previous behavior" here because in time it won't be relevant anymore.  The release notes would be a better place to compare old and new behavior.

Author: Yes, sure. Thank you. Done.

See [action.yml](action.yml)

## Basic:
```yaml
steps:
  - uses: actions/checkout@v2
  - uses: actions/setup-go@v2
    with:
      go-version: '1.16.1' # The Go version to download (if necessary) and use.
  - run: go run hello.go
```

## Check latest version:  

The `check-latest` flag defaults to `false`. Use the default or set `check-latest` to `false` if you prefer stability and if you want to ensure a specific Go version is always used.

> If `check-latest` is set to `true`, the action first checks if the cached version is the latest one. If the locally cached version is not the most up-to-date, a Go version will then be downloaded. Set `check-latest` to `true` it you want the most up-to-date Go version to always be used.

Reviewer 2: ```suggestion
If `check-latest` is set to `true`, the action first checks if the cached version is the latest one. If the locally cached version is not the most up-to-date, a Go version will then be downloaded. Set `check-latest` to `true` if you want the most up-to-date Go version to always be used.
```

Reviewer 2: typo (it -> if)

Author: Done. Thank you.

> Setting `check-latest` to `true` has performance implications as downloading Go versions is slower than using cached versions.

```yaml
steps:
  - uses: actions/checkout@v2
  - uses: actions/setup-go@v2
    with:
      go-version: '1.14'
      check-latest: true
  - run: go run hello.go
```

> ## Matrix Testing:  

Reviewer 2: ```suggestion
## Matrix Testing
```
Thanks for making this a heading 👍

Author: Done. Thank you.
